File: backend/src/main/java/org/example/backend/domain/fish/repository/FishRepository.java
package org.example.backend.domain.fish.repository;

import java.util.List;
import java.util.Optional;
import org.example.backend.domain.fish.entity.Fish;
import org.springframework.data.jpa.repository.JpaRepository;

public interface FishRepository extends JpaRepository<Fish, Long> {

  long countByAquarium_Id(Long aquariumId);

  List<Fish> findAllByAquarium_Id(Long aquariumId);

  Optional<Fish> findByAquarium_IdAndId(Long aquariumId, Long fishId);
}

File: backend/src/main/java/org/example/backend/domain/fish/repository/FishRepository.kt
package org.example.backend.domain.fish.repository

import org.example.backend.domain.fish.entity.Fish
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface FishRepository : JpaRepository<Fish, Long> {
    fun countByAquarium_Id(aquariumId: Long): Long

    fun findAllByAquarium_Id(aquariumId: Long): List<Fish>

    fun findByAquarium_IdAndId(aquariumId: Long, fishId: Long): Fish?
}